mongo/integration/unified_spec_test.go

```diff
@@ -437,6 +437,19 @@ func executeTestRunnerOperation(mt *mtest.T, testCase *testCase, op *operation,
 		fp, err := op.Arguments.LookupErr("failPoint")
 		assert.Nil(mt, err, "failPoint not found in arguments")
 		mt.SetFailPointFromDocument(fp.Document())
+	case "assertSessionTransactionState":
+		stateVal, err := op.Arguments.LookupErr("state")
+		assert.Nil(mt, err, "state not found in arguments")
+		expectedState, ok := stateVal.StringValueOK()
+		assert.True(mt, ok, "state argument is not a string")
+		actualState := clientSession.TransactionState.String()
```

Comment:
Nit: should this also assert `clientSession` is not `nil`?

Reply:
That's a good idea. I think the only time `clientSession` would be `nil` is if there was a malformed test with an `assertSessionTransactionState` and no corresponding `session` argument, but it's still worth having a check. There are numerous assertions on the fields of `clientSession` in this function without preceding `NotNil` assertions on `clientSession`, so I filed GODRIVER-2117.